org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/ClasspathMultiDirectory.java

```diff
@@ -33,14 +34,16 @@ public class ClasspathMultiDirectory extends ClasspathDirectory {
 char[][] exclusionPatterns; // used by builders when walking source folders
 boolean hasIndependentOutputFolder; // if output folder is not equal to any of the source folders
 public boolean ignoreOptionalProblems;
+int release; //if given this sets an explicit release level for this directory overriding project settings
 
 ClasspathMultiDirectory(IContainer sourceFolder, IContainer binaryFolder, char[][] inclusionPatterns, char[][] exclusionPatterns,
-		boolean ignoreOptionalProblems, IPath externalAnnotationPath) {
+		boolean ignoreOptionalProblems, IPath externalAnnotationPath, int release) {
 	super(binaryFolder, true, null, externalAnnotationPath, false /* source never an automatic module*/);
 
 	this.sourceFolder = sourceFolder;
 	this.inclusionPatterns = inclusionPatterns;
 	this.exclusionPatterns = exclusionPatterns;
+	this.release = release>8?release:IReleaseAwareNameEnvironment.DEFAULT_RELEASE;
```

Comment:
let's avoid the magic number `8`, please.

Reply:
Done!